llvm/include/llvm/IR/DebugInfo.h

```diff
@@ -314,6 +317,25 @@ class AssignmentTrackingPass : public PassInfoMixin<AssignmentTrackingPass> {
 
 /// Return true if assignment tracking is enabled for module \p M.
 bool isAssignmentTrackingEnabled(const Module &M);
+
+template <> struct DenseMapInfo<at::VarRecord> {
+  static inline at::VarRecord getEmptyKey() {
+    return at::VarRecord{nullptr, nullptr};
+  }
+
+  static inline at::VarRecord getTombstoneKey() {
+    return at::VarRecord{nullptr, nullptr};
+  }
```

Comment:
The tombstone key and empty key mustn't be equal as this breaks the implementation of `DenseMap`. It can lead to accidently finding a member on lookup when it shouldn't or not finding a member when it should.

What you can do is use e.g. `DenseMapInfo<DILocalVariable*>::getTombstoneKey()` as a value here and the corresponding empty key in the other.

Reply:
Thank you. I was 90% sure I was doing this wrong.